up42/catalog.py

```diff
@@ -283,3 +285,114 @@ def download_quicklooks(
+        """
+        Helper that constructs necessary parameters for `Order.place` and `Order.estimate`.
+
+        Args:
+            geometry (Union[ Dict, Feature, FeatureCollection, List, GeoDataFrame, Point, Polygon, ]): The intended
+                output AOI of the order.
+            scene (GeoSeries): A single item/row of the result of `Catalog.search`. For instance, search_results.loc[0]
+                for the first scene of a catalog search result.
+
+        Returns:
+            str, Dict: A tuple including a provider name and order parameters.
+        """
+        if not isinstance(scene, Series):
+            raise ValueError(
+                "`scene` parameter must be a GeoSeries, or a single item/row of a GeoDataFrame. "
+                "For instance, search_results.loc[0] returns a GeoSeries."
+            )
+        aoi_fc = any_vector_to_fc(
+            vector=geometry,
+        )
+        aoi_geometry = fc_to_query_geometry(
+            fc=aoi_fc,
+            geometry_operation="intersects",
+            squash_multiple_features="union",
+        )
+        data_provider_name = scene.providerName
+        order_params = {"id": scene.id, "aoi": aoi_geometry}
+        return data_provider_name, order_params
+
+    def estimate_order(
+        self,
+        geometry: Union[
+            Dict,
+            Feature,
+            FeatureCollection,
+            List,
+            GeoDataFrame,
+            Point,
+            Polygon,
+        ],
+        scene: Series,
+    ) -> int:
+        """
+        Estimate the cost of an order from an item/row in a result of `Catalog.search`.
+
+        Args:
+            geometry (Union[ Dict, Feature, FeatureCollection, List, GeoDataFrame, Point, Polygon, ]): The intended
+                output AOI of the order.
+            scene (GeoSeries): A single item/row of the result of `Catalog.search`. For instance, search_results.loc[0]
+                for the first scene of a catalog search result.
+
+        Returns:
+            int: An estimated cost for the order in UP42 credits.
+        """
+        data_provider_name, order_params = self._order_payload(geometry, scene)
+        return Order.estimate(self.auth, data_provider_name, order_params)
+
+    def place_order(
+        self,
+        geometry: Union[
+            Dict,
+            Feature,
+            FeatureCollection,
+            List,
+            GeoDataFrame,
+            Point,
+            Polygon,
+        ],
+        scene: Series,
+        track_status: bool = False,
+        report_time: int = 120,
+    ) -> "Order":
+        """
+        Place an order from an item/row in a result of `Catalog.search`.
+
+        Args:
+            geometry (Union[ Dict, Feature, FeatureCollection, List, GeoDataFrame, Point, Polygon, ]): The intended
+                output AOI of the order.
+            scene (GeoSeries): A single item/row of the result of `Catalog.search`. For instance, search_results.loc[0]
```

Comment:
Maybe mention the required columns or assert even?

Love dataframes and having to provide the provider_id and correct provider explicitly probably leads to problems, but the dataframe row also doesn't feel right, especially since it's a single item only. Tough. Guess still best solution. Will play around a bit with this when merged before release.

Reply:
Not sure I fully understand what you mean - are you saying you expect users to build up their geodataframe/series from scratch and pass it on to the method? 

To create an order we need a provider and a catalog item id (plus the geometry that comes from the user). The catalog item id comes only from the catalog search (i.e. you can't get it anywhere else as far as I know).

My understanding was that we made a conscious decision to return a GeoDataFrame (by default) when we perform a catalog search. With that, following methods that require "search results" should also handle either a GeoDataFrame or some child of it. In this case since an order can only be placed for a single "search result" I would argue it's fair to say we require a row of this data frame. In the future we could also think of making some method to iterate over "search results" and place orders, but I feel that is premature right now. 

Toughts?

Reply:
No, I agree with that giving the dataframe row is the best option here, and it is the default result from catalog search anyway. But since you have to provide only a single dataframe row, and giving a dataframe row as an argument is a bit unusual, I was thinking if it would maybe be more explicit to just have the user provide the `provider_id` and `provider` explicitly. But that would lead to other problems (user provides the scene_id etc. because he doesn't know the difference), and I also missed that the geometry is taken from the dataframe row.

So in the end agree with this solution, just wanted to communicate some concerns, maybe you had an idea for another solution etc.